code/modules/events/spider_terror.dm

```diff
@@ -44,11 +43,18 @@
 			// Strongest, only used during highpop.
 			spider_type = /mob/living/simple_animal/hostile/poison/terror_spider/queen
 			spawncount = 1
-	while(spawncount && length(vents))
-		var/obj/vent = pick_n_take(vents)
-		new spider_type(vent.loc)
-		spawncount--
-		successSpawn = TRUE
+	spawn() // Just like in xeno and demon events, this spawn() is necessary. Do not remove.
```

Comment:
I am aware that killing the `spawn()` here will break things. **However**, I am going to ask you offload this to another proc and use `INVOKE_ASYNC` to do the work instead of wrapping all of this inside a `spawn()`